distributions/aws/test/e2e/tests/test_rds_s3.py

```diff
@@ -0,0 +1,444 @@
+
+        # Disable caching in KFP
+        # By default KFP will cache previous pipeline runs and subsequent runs will skip cached steps
+        # This prevents artifacts from being uploaded to s3 for subsequent runs
+        patch_body = unmarshal_yaml(DISABLE_PIPELINE_CACHING_PATCH_FILE)
+        k8s_admission_registration_api_client = (
+            create_k8s_admission_registration_api_client(cluster, region)
+        )
+        k8s_admission_registration_api_client.patch_mutating_webhook_configuration(
+            "cache-webhook-kubeflow", patch_body
+        )
+
+        metadata_file = metadata.to_file()
+        print(metadata.params)  # These needed to be logged
+        print("Created metadata file for TestRDSS3", metadata_file)
+
+    def test_kfp_experiment(self, setup, kfp_client, cfn_stack, cfn_client):
+        stack_outputs = get_stack_outputs(cfn_client, cfn_stack["stack_name"])
+
+        name = rand_name("experiment-")
+        description = rand_name("description-")
+        experiment = kfp_client.create_experiment(
+            name, description=description, namespace=DEFAULT_USER_NAMESPACE
+        )
+
+        assert name == experiment.name
+        assert description == experiment.description
+        assert DEFAULT_USER_NAMESPACE == experiment.resource_references[0].key.id
+
+        mysql_client = create_mysql_client(
+            user=cfn_stack["params"]["DBUsername"],
+            password=cfn_stack["params"]["DBPassword"],
+            host=stack_outputs["RDSEndpoint"],
+            database="mlpipeline",
+        )
+
+        resp = mysql.query(
+            mysql_client, f"select * from experiments where Name='{name}'"
+        )
+        assert len(resp) == 1
+        assert resp[0]["Name"] == experiment.name
+        assert resp[0]["Description"] == experiment.description
+        assert resp[0]["Namespace"] == experiment.resource_references[0].key.id
+
+        resp = kfp_client.get_experiment(
+            experiment_id=experiment.id, namespace=DEFAULT_USER_NAMESPACE
+        )
+
+        assert name == resp.name
+        assert description == resp.description
+        assert DEFAULT_USER_NAMESPACE == resp.resource_references[0].key.id
+
+        kfp_client.delete_experiment(experiment.id)
+
+        resp = mysql.query(
+            mysql_client, f"select * from experiments where Name='{name}'"
+        )
+        assert len(resp) == 0
+
+        try:
+            kfp_client.get_experiment(
+                experiment_id=experiment.id, namespace=DEFAULT_USER_NAMESPACE
+            )
+            raise AssertionError("Expected KFPApiException Not Found")
+        except KFPApiException as e:
+            assert "Not Found" == e.reason
+
+        mysql_client.close()
+
+    def test_run_pipeline(self, setup, kfp_client, cfn_stack, cfn_client, s3_client):
+        stack_outputs = get_stack_outputs(cfn_client, cfn_stack["stack_name"])
+
+        experiment_name = rand_name("experiment-")
```

Comment:
it will be good to move some of the pipeline and katib job run part to a common method so other tests can reuse it. and also reuse b/w generic and this test

not a blocker for this PR but will be nice if we can have it

Reply:
will add a todo for now

Reply:
or somehow use `pytest.parametrize` https://github.com/kubeflow/pipelines/blob/master/components/aws/sagemaker/tests/integration_tests/component_tests/test_train_component.py#L16